src/dcqc/suites/suite_abc.py

```diff
@@ -207,20 +216,22 @@ def compute_tests(self) -> None:
         for test in self.tests:
             test.get_status()
 
-    def compute_status(self) -> TestStatus:
+    def compute_status(self) -> SuiteStatus:
         """Compute the overall suite status."""
         self.compute_tests()
-        if self._status != TestStatus.NONE:
+        if self._status != SuiteStatus.NONE:
             return self._status
-        self._status = TestStatus.PASS
+        self._status = SuiteStatus.GREEN
         for test in self.tests:
             test_name = test.type
-            if test_name not in self.required_tests:
-                continue
             test_status = test.get_status()
-            self._status = test_status
-            if self._status == TestStatus.FAIL:
-                break
+            if test_name in self.required_tests:
+                if test_status == TestStatus.FAIL:
+                    self._status = SuiteStatus.RED
+                    break
```

Comment:
Nit: You could also return `self._status` here.

Reply:
changed in the latest commit